atc/exec/put_step.go

```diff
@@ -109,7 +110,7 @@ func (step *PutStep) Run(ctx context.Context, state RunState) error {
 		putInputs = NewSpecificInputs(step.plan.Inputs.Specified)
 	}
```

Comment:
I know this code is not new, but we might have an opportunity here to refactor this confusing control flow - I just caught it because the implementation of specific inputs and inputs is changing. Maybe one for the refactoring club!! @pivotal-jamie-klassen